exploration/code-mode-introducer.md

```diff
@@ -0,0 +1,224 @@
+# Design Proposal: Choosing a Code Mode Introducer
+
+Status: **Proposed**
+
+<details>
+	<summary>Metadata</summary>
+	<dl>
+		<dt>Contributors</dt>
+		<dd>@aphillips</dd>
+		<dt>First proposed</dt>
+		<dd>2023-11-10</dd>
+		<dt>Pull Requests</dt>
+		<dd>#000</dd>
+	</dl>
+</details>
+
+## Objective
+
+_What is this proposal trying to achieve?_
+
+It must be possible to reliably parse messages.
+
+Our current syntax features unquoted patterns for simple messages
+and unquoted code tokens with quoted patterns for complex messages.
+Determining whether a message will have code tokens requires some
+special character sequence, either part of the code itself or
+prepended to the message.
```

Comment:
```suggestion
For ease of parsing, a distinguished character sequence
should denote the presence of code tokens in a message
(either part of the code itself or prepended to the message).
```

(Strictly speaking, it doesn't require that, since you could just scan for the first code token -- but that's undesirable since it requires arbitrary lookahead.)